pilot/pkg/model/sidecar.go

```diff
@@ -38,12 +38,12 @@ var (
 		gvk.ServiceEntry:    {},
 		gvk.VirtualService:  {},
 		gvk.DestinationRule: {},
+		gvk.Sidecar:         {},
```

Comment:
Hold, what's the expectation if the sidecar workload selector changed? Say if sidecar A related to proxy A, and then we changed the sidecar A(not selecting A) anymore.

Reply:
In this case, we can get the `Sidecar` config from the `proxy.PrevSidecarScope` and determine if it is affected.

In summary, we hold two kinds of information: the previous `SidecarScope`/`Sidecar` that affects the proxy; and the `SidecarScope`/`Sidecar` currently affecting the proxy. And the `Sidecar` config associated with that proxy, will be only one of the two. 
So we won't lose any changes to the `Sidecar` config that might affect the proxy.

Translated with www.DeepL.com/Translator (free version)

Reply:
LG, this is handled by 
```
		if proxy.SidecarScope.DependsOnConfig(config) {
			return true
		} else if proxy.PrevSidecarScope != nil && proxy.PrevSidecarScope.DependsOnConfig(config) {
			return true
		}
```